packages/rum/src/domain/record/serialize.ts

```diff
@@ -356,22 +356,33 @@ function getValidTagName(tagName: string): string {
   return processedTagName
 }
 
-function getCssRulesString(s: CSSStyleSheet): string | null {
-  try {
-    const rules = s.rules || s.cssRules
-    if (rules) {
-      const styleSheetCssText = Array.from(rules, getCssRuleString).join('')
-      return switchToAbsoluteUrl(styleSheetCssText, s.href)
-    }
+function getCssRulesString(cssStyleSheet: CSSStyleSheet | undefined | null): string | null {
+  // TODO: remove monitor once we're confident the try/catch was not hiding more error
+  return monitor(_getCssRulesString)(cssStyleSheet)
+}
 
+function _getCssRulesString(cssStyleSheet: CSSStyleSheet | undefined | null): string | null {
+  if (!cssStyleSheet) {
     return null
-  } catch (error) {
-    return null
+  }
+  try {
+    const rules = cssStyleSheet.rules || cssStyleSheet.cssRules
+    if (!rules) {
+      return null
+    }
+    const styleSheetCssText = Array.from(rules, getCssRuleString).join('')
+    return switchToAbsoluteUrl(styleSheetCssText, cssStyleSheet.href)
+  } catch (err) {
+    if (err instanceof DOMException && err.name === 'SecurityError') {
+      // if css is protected by CORS we cannot access cssRules see: https://www.w3.org/TR/cssom-1/#the-cssstylesheet-interface
+      return null
+    }
+    throw err
   }
 }
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏I would reduce the `try/catch` scope to be minimal, and remove the error instance check:

```ts
function _getCssRulesString(cssStyleSheet: CSSStyleSheet | undefined | null): string | null {
  if (!cssStyleSheet) {
    return null
  }
  let rules
  try {
    rules = cssStyleSheet.rules || cssStyleSheet.cssRules
  } catch {
    // if css is protected by CORS we cannot access cssRules see: https://www.w3.org/TR/cssom-1/#the-cssstylesheet-interface
  }
  if (!rules) {
    return null
  }
  const styleSheetCssText = Array.from(rules, getCssRuleString).join('')
  return switchToAbsoluteUrl(styleSheetCssText, cssStyleSheet.href)
}
```

Reply:
@bcaudan what do you think? Is it worth logging unexpected exceptions here to be aware of things that are failing?

Reply:
> Is it worth logging unexpected exceptions here to be aware of things that are failing?

ultimately, if accessing those `rules` or `cssRules` raises an exception, we would not be able to do much.
It could be interesting to see if there are other cases than the ones identified (for curiosity sake) but eventually we should probably just leave the catch clause empty.